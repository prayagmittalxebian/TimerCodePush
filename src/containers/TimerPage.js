import React, { Component } from 'react';
import { Text, FlatList, View, StyleSheet, Button } from 'react-native';
import { connect } from 'react-redux'
import TimerList from './../components/TimerList/TimerList';
import * as timerActionCreator from './../actions/timerAction';

class TimerPage extends Component {

    constructor(props) {
        super(props);
    }

    startTimer = () => {
        this.props.dispatch(timerActionCreator.startTimer())
        this.timerInterval = setInterval(() => { this.props.dispatch(timerActionCreator.updateTimer()); }, 1000)
    }

    stopTimer = () => {
        let timerList = this.props.time.timerList;
        console.log(":timerList", timerList)
        timerList.push(this.props.time.seconds);
        console.log("update", timerList)
        this.props.dispatch(timerActionCreator.stopTimer(timerList))
        clearInterval(this.timerInterval);
    }

    resetTimer = () => {
        this.props.dispatch(timerActionCreator.resetTimer())
        clearInterval(this.timerInterval);
    }

    render() {
        return (
            <View style={styles.container}>
                <View style={styles.actionContainer}>
                    <View style={styles.buttonView}>
                        {
                            this.props.time.isTimerStarted ?
                                <Button title="Stop" onPress={this.stopTimer}></Button> :
                                <Button title="Start" onPress={this.startTimer}></Button>
                        }
                    </View>
                    <View style={styles.timer}>
                        <Text>{this.props.time.seconds}</Text>
                    </View>
                    <View style={[styles.buttonView]}>
                        <Button title="Reset" onPress={this.resetTimer}></Button>
                        <Button title="Record"></Button>
                    </View>
                </View>
                <View style={styles.listContainer}>
                    <TimerList timerList={this.props.time.timerList} />
                </View>
            </View>
        );
    }
}

const mapStateToProps = (state) => ({
    time: state.time
});

export default connect(mapStateToProps)(TimerPage)

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "lightgrey"
    },
    actionContainer: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: "space-between"
    },
    listContainer: {
        flex: 1
    },
    buttonView: {
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    resetButtonContainer: {
        marginTop: 10
    },
    timer: {
        justifyContent: 'center',
        alignItems: "center"
    }
});