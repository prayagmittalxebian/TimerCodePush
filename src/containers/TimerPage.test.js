import React from "react";
import TimerPage from "./TimerPage";
import { shallow, mount } from "enzyme";
import thunk from "redux-thunk";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";


const state = {
  time: {
    timerList: [],
    isTimerStarted: false,
    seconds: 0
  }
}
const mockStore = configureStore([thunk]);
const store = mockStore(state);

describe("Timer Transaction Row", () => {
  const snapshotProps = {

  };

  it("Linked Transaction Row SnapShot Test", () => {
    const tree = renderer
      .create(<TimerPage {...snapshotProps} store={store} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});