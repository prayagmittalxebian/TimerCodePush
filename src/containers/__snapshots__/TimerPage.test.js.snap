// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer Transaction Row Linked Transaction Row SnapShot Test 1`] = `
<View
  style={
    Object {
      "backgroundColor": "lightgrey",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Start
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text>
        0
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Reset
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Record
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <RCTScrollView
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</View>
`;
